lib/index.js

```diff
@@ -44,20 +44,43 @@ export default function remarkTextr(options) {
   const settings = options || emptyOptions
   const plugins = settings.plugins || []
   const textrOptions = settings.options || {}
-  const promise = Promise.all(
-    plugins.map(async function (fn) {
-      if (typeof fn === 'string') {
-        // Assume plugin at default.
-        /** @type {{default: TextrPlugin}} */
-        const mod = await import(fn)
-        return mod.default
-      }
 
-      return fn
-    })
-  ).then(function (list) {
-    return textr(textrOptions).use(...list)
-  })
+  const noNeedPromise =
+    plugins.filter((plugin) => typeof plugin === 'string').length === 0
+
+  const promise = noNeedPromise
```

Comment:
I recommend an if/else statement here, and exiting early

```
if (needsPromise)	{
  const typography = textr(textrOptions).use(...plugins)
  return function (tree) {
    visit(tree, 'text', function (node) {
      node.value = typography.exec(node.value)
    })
  }
}

// rest of code awaiting promise
```

Reply:
I applied.